bundles/org.openhab.binding.pushover/src/main/java/org/openhab/binding/pushover/internal/connection/PushoverMessageBuilder.java

```diff
@@ -0,0 +1,267 @@
+        }
+
+        return new PushoverMessageBuilder(apikey, user);
+    }
+
+    public PushoverMessageBuilder withMessage(String message) {
+        this.message = message;
+        return this;
+    }
+
+    public PushoverMessageBuilder withTitle(String title) {
+        this.title = title;
+        return this;
+    }
+
+    public PushoverMessageBuilder withDevice(String device) {
+        this.device = device;
+        return this;
+    }
+
+    public PushoverMessageBuilder withPriority(int priority) {
+        this.priority = priority;
+        return this;
+    }
+
+    public PushoverMessageBuilder withRetry(int retry) {
+        this.retry = retry;
+        return this;
+    }
+
+    public PushoverMessageBuilder withExpire(int expire) {
+        this.expire = expire;
+        return this;
+    }
+
+    public PushoverMessageBuilder withUrl(String url) {
+        this.url = url;
+        return this;
+    }
+
+    public PushoverMessageBuilder withUrlTitle(String urlTitle) {
+        this.urlTitle = urlTitle;
+        return this;
+    }
+
+    public PushoverMessageBuilder withSound(String sound) {
+        this.sound = sound;
+        return this;
+    }
+
+    public PushoverMessageBuilder withAttachment(String attachment) {
+        this.attachment = attachment;
+        return this;
+    }
+
+    public PushoverMessageBuilder withContentType(String contentType) {
+        this.contentType = contentType;
+        return this;
+    }
+
+    public PushoverMessageBuilder withHtmlFormatting() {
+        this.html = true;
+        return this;
+    }
+
+    public PushoverMessageBuilder withMonospaceFormatting() {
+        this.monospace = true;
+        return this;
+    }
+
+    public ContentProvider build() {
+        if (message != null) {
+            if (message.length() > MAX_MESSAGE_LENGTH) {
+                throw new IllegalArgumentException(String.format(
+                        "Skip sending the message as 'message' is longer than %d characters.", MAX_MESSAGE_LENGTH));
+            }
+            body.addFieldPart(MESSAGE_KEY_MESSAGE, new StringContentProvider(message), null);
+        }
+
+        if (title != null) {
+            if (title.length() > MAX_TITLE_LENGTH) {
+                throw new IllegalArgumentException(String
+                        .format("Skip sending the message as 'title' is longer than %d characters.", MAX_TITLE_LENGTH));
+            }
+            body.addFieldPart(MESSAGE_KEY_TITLE, new StringContentProvider(title), null);
+        }
+
+        if (device != null) {
+            if (device.length() > MAX_DEVICE_LENGTH) {
+                logger.warn("Skip 'device' as it is longer than {} characters. Got: {}.", MAX_DEVICE_LENGTH, device);
+            } else {
+                // TODO [A-Za-z0-9_-]
+                // TODO Messages may be addressed to multiple specific devices by joining them with a comma (such as
+                // device=iphone,nexus5)
```

Comment:
Can this TODO be done?

Reply:
Not needed anymore. Can be passed by parameter directly.